File: backend/java-core-api/src/main/java/com/aifashionstudio/ordering/application/command/UpdateOrderStatusCommand.java
package com.aifashionstudio.ordering.application.command;

import com.aifashionstudio.ordering.domain.model.OrderStatus;

import java.util.UUID;

public record UpdateOrderStatusCommand(
        UUID staffId,
        UUID orderId,
        OrderStatus toStatus,
        String note
) {
}

File: backend/dotnet-platform-api/AiFashionStudio.Platform.Application/Staff/Commands/UpdateOrderStatus/UpdateOrderStatusCommand.cs
using AiFashionStudio.Platform.Application.Common.Dtos;
using MediatR;

namespace AiFashionStudio.Platform.Application.Staff.Commands.UpdateOrderStatus;

/// <summary>
/// Staff yêu cầu chuyển trạng thái đơn hàng. Gateway chuyển tiếp sang Java Order Service —
/// Java giữ state machine và ghi order_status_history.
/// </summary>
public record UpdateOrderStatusCommand(
    Guid OrderId,
    string ToStatus,
    string? Note,
    Guid ChangedBy) : IRequest<UpdateOrderStatusResponse>;
